components/automate-gateway/gateway/services.go

```diff
@@ -672,18 +672,25 @@ func (s *Server) ReportExportHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	writeContent(w, stream, query.Type)
+	writeContent(w, stream, query.Type, "ids")
 }
 
-func writeContent(w http.ResponseWriter, stream reporting.ReportingService_ExportClient, queryType string) {
-	//when the type is json, we will be retrieving a json array of objects and since we will be getting them one at a
-	// time, we need to provide the '[' to open and the ']' to close (the close will happen on EOF, below)
+func writeContent(w http.ResponseWriter, stream reporting.ReportingService_ExportClient, queryType string, wrapper string) {
+	//we will be retrieving an array of objects and since we will be getting them one at a time, we need to provide the
+	//opening and closing elements to close for when we are dealing with json or xml (the close will happen on EOF, below)
+	//csv has no wrapping element so there is no element to prepend or append for csv type
 	if queryType == "json" {
 		_, err := w.Write([]byte("["))
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
+	} else if queryType == "xml" {
+		_, err := w.Write([]byte(fmt.Sprintf("<%s>\n", wrapper)))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
```

Comment:
Nice, but could (a) be a bit more concise and (b) show the intent more clearly:
```suggestion
	openBracket := nil
	closeBracket := nil
	switch queryType {
	case "json":
		openBracket = "["
		closeBracket = "]"
	case "xml":
		openBracket = fmt.Sprintf("<%s>\n", wrapper)
		closeBracket = fmt.Sprintf("\n</%s>", wrapper)
	}
	if openBracket != nil {
		_, err := w.Write([]byte(openBracket))
		if err != nil {
			http.Error(w, err.Error(), http.StatusInternalServerError)
			return
		}
	}
	for {
		data, err := stream.Recv()
		if err == io.EOF {
			if closeBracket != nil {
				_, err = w.Write([]byte(closeBracket))
				if err != nil {
					http.Error(w, err.Error(), http.StatusInternalServerError)
					return
				}
			}
```

Reply:
I l